src/upgrade.rs

```diff
@@ -21,7 +21,7 @@ pub trait UpgradeHook {
 /// Naked upgrade function which calls migrate method on the contract
 pub fn upgrade<T>()
 where
-    T: BorshDeserialize + BorshSerialize,
+    T: UpgradeHook + ?Sized,
```

Comment:
Does this function need generic parameters at all?

Reply:
I think generic parameters can be removed